common/static/common/js/discussion/views/new_post_view.js

```diff
@@ -100,6 +109,26 @@
                 }
             };
 
+            NewPostView.prototype.getGroupName = function() {
+                var userGroupId;
+                var cohort;
+                var group_name = null;
+                if (this.course_settings.get('is_cohorted')) {
+                    userGroupId = $('#discussion-container').data('user-group-id');
+                    if (!userGroupId) {
+                        userGroupId = this.user_group_id;
```

Comment:
This is due to the different "modes" the discussion can work under.  Either in the discussion tab ("tab mode"), in which only the data attribute is available, or in a course unit as a module, where  the user's group ID wasn't available at all (had to add it to the JSON response).